compiler/rustc_trait_selection/src/traits/fulfill.rs

```diff
@@ -352,7 +352,7 @@ impl<'a, 'b, 'tcx> FulfillProcessor<'a, 'b, 'tcx> {
                 // Evaluation will discard candidates using the leak check.
                 // This means we need to pass it the bound version of our
                 // predicate.
-                ty::PredicateKind::Trait(trait_ref, _constness) => {
+                ty::PredicateKind::Trait(trait_ref) => {
                     let trait_obligation = obligation.with(binder.rebind(trait_ref));
```

Comment:
This is the main change of this PR, did the other changes affect tests as well, or is it just this change?

Reply:
No, this is the main change:
https://github.com/rust-lang/rust/blob/8c2ed7012383ecca11d40c621604e8d30c329873/compiler/rustc_trait_selection/src/traits/select/mod.rs#L1025-L1038

Reply:
hmm... I would expect this change to have some effect, too.